### PR TITLE
build.gradle: Migrate from maven to maven-publish plugin.

### PR DESCRIPTION
**Pull request by schildbach**
To publish to the local Maven repository, use `gradle publishToMavenLocal` rather than `gradle install`.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.10'
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.16'
```
Comment (schildbach) [inline build.gradle:7]: I wonder is it possible to apply different versions on the condition of Gradle version? @msgilligan

Comment (msgilligan) [reply build.gradle:7]: I might be able to do that...

**Comment by msgilligan** [comment]
See PR #2134 and PR #2135

File: build.gradle (hunk 1 above)
Comment (msgilligan) [reply build.gradle:7]: @schildbach See: https://github.com/bitcoinj/bitcoinj/pull/2140

**Comment by schildbach** [comment]
Merged the maven change via 8a775a37d8d2f7edb0b7b9cf5c615867062f0f5e, and dropped the protobuf change in favor of #2140.

**Pull request closed without merging**